Allow `optional<bool>` as action parameter

Getting an error whenever I enter an `optional<bool>` however it does work for other EOSIO types such as `optional<name>`

This seems like a valid syntax in C++
https://www.boost.org/doc/libs/1_56_0/libs/optional/doc/html/boost_optional/tutorial/a_note_about_optional_bool_.html

## Code example

```c++
#include <eosio/eosio.hpp>
#include <optional>

using namespace eosio;
using namespace std;

class [[eosio::contract]] hello : public contract {
  public:
      using contract::contract;

      [[eosio::action]]
      void hi( optional<bool> option ) {
         if ( option == true ) print( "print me" );
      }
};
```

## ❌ error message

```bash
/var/folders/sc/k5bwq66j0g16dn721lwhvf540000gn/T//hello.cpp:32:15: error: use of undeclared identifier '_Bool'
std::optional<_Bool> arg1; ds >> arg1;
              ^
/var/folders/sc/k5bwq66j0g16dn721lwhvf540000gn/T//hello.cpp:32:22: error: C++ requires a type specifier for all
      declarations
std::optional<_Bool> arg1; ds >> arg1;
                     ^
2 errors generated.
```

## expected ABI

```json
{
    "name": "hi",
    "base": "",
    "fields": [
        {
            "name": "option",
            "type": "bool?"
        }
    ]
}
```

## Inline `optional<bool>` does work ✅ 

```c++
[[eosio::action]]
void hi( ) {
  optional<bool> option = true;
  if ( option == true ) print( "print me" );
}
```

@jeffreyssmith2nd strange that 1.7 works for you

I'm on MacOS Catalina, pretty sure that doesn't have any impact.

Are you attempting to compile?

```c++
#include <eosio/eosio.hpp>
#include <optional>

using namespace eosio;
using namespace std;

class [[eosio::contract]] hello : public contract {
  public:
      using contract::contract;

      [[eosio::action]]
      void hi( optional<bool> option ) {
         if ( option == true ) print( "print me" );
      }
};
```

Still getting an error with `eosio-cpp version 1.7.0`

```
$ eosio-cpp hello.cpp -o hello.wasm
Warning, empty ricardian clause file
Warning, empty ricardian clause file
Warning, action <hi> does not have a ricardian contract
/var/folders/sc/k5bwq66j0g16dn721lwhvf540000gn/T//hello.cpp:31:15: error: use of undeclared identifier '_Bool'
std::optional<_Bool> arg0; ds >> arg0;
              ^
/var/folders/sc/k5bwq66j0g16dn721lwhvf540000gn/T//hello.cpp:31:22: error: C++ requires a type specifier for all declarations
std::optional<_Bool> arg0; ds >> arg0;
                     ^
2 errors generated.
/usr/local/Cellar/eosio.cdt/1.7.0/opt/eosio.cdt/bin/wasm-ld: error: cannot open /var/folders/sc/k5bwq66j0g16dn721lwhvf540000gn/T//hello.cpp.o: No such file or directory
```

Tried reinstalling & installing CDT on MacOS, still getting the same error.

I also installed the latest CDT 1.7.0 on Debian "GNU/Linux 10 (buster)" and getting similar error:

https://github.com/eosio/eosio.cdt#debian-package-install

```bash
$ cat /etc/debian_version
10.3

$ eosio-cpp --version
eosio-cpp version 1.7.0

$ eosio-cpp hello.cpp -o hello.wasm
Warning, empty ricardian clause file
Warning, empty ricardian clause file
Warning, action <hi> does not have a ricardian contract
/tmp/hello.cpp:31:15: error: use of undeclared identifier '_Bool'
std::optional<_Bool> arg0; ds >> arg0;
              ^
/tmp/hello.cpp:31:22: error: C++ requires a type specifier for all declarations
std::optional<_Bool> arg0; ds >> arg0;
                     ^
2 errors generated.
/usr/opt/eosio.cdt/1.7.0/bin/wasm-ld: error: cannot open /tmp/hello.cpp.o: No such file or directory
```

--- a/tools/include/eosio/codegen.hpp
+++ b/tools/include/eosio/codegen.hpp
@@ -185,6 +185,14 @@ namespace eosio { namespace cdt {
             return true;
          }
 
+         void normalize_type_name(std::string& type_name) {
+            // Currently _Bool is the only one of these oddities but others could be handled here if they arise.
+            auto pos = type_name.find("_Bool");
+            if (pos != std::string::npos) {
+               type_name.replace(pos, 5, "bool");
+            }
+         }
+
          template <size_t N>
          void emitError(CompilerInstance& inst, SourceLocation loc, const char (&err)[N]) {
             FullSourceLoc full(loc, inst.getSourceManager());
@@ -268,7 +276,7 @@ namespace eosio { namespace cdt {
                   qt.removeLocalVolatile();
                   qt.removeLocalRestrict();
                   std::string tn = clang::TypeName::getFullyQualifiedName(qt, *(cg.ast_context), policy);
-                  tn = tn == "_Bool" ? "bool" : tn; // TODO look out for more of these oddities
+                  normalize_type_name(tn);
                   ss << tn << " arg" << i << "; ds >> arg" << i << ";\n";
                   i++;
                }

